### PR TITLE
Request for twigacloud_signup app approval

### PR DESCRIPTION
**Pull request by brunoalfred**
### Link to the App

- https://github.com/brunoalfred/nextcloud_signup

The App is in active development.

### CODE REVIEW

**Comment by nickvergessen** [comment]
Apart from the phone number verification this sounds similar to the registration app:
https://github.com/nextcloud/registration

Want to join forces there?

**Comment by brunoalfred** [comment]
> 
We are planning to support phone number registration for hosted instance in an organization, perhaps SMS will be sent using the internal API(private) for security and privacy policy. 

Also registration app works  just fine, the only challenge is that it support verification through email and if it were to support SMS to verify users then it won't be able to accommodate all APIs/Services used to send SMS.

Perhaps i didn't find a way to develop private apps.

**Comment by nickvergessen** [comment]
That's okay then.

It still makes sense to register an appid, so your private app is not overwriten by an update from the appstore at some time. However since this would block quite a keyword for a private app, maybe you want to use something more unique instead if okay prefixed with the org name, your name or a fantasy name? Because if it is a private app no one will ever see it (apart from this repository)

**Comment by brunoalfred** [comment]
> 
- Do i have to regenerate everything and PR once more? 
- Planning to user twigacloud_singup for APP ID, is it ok? 
- can i keep the repo private just incase?
- How are private apps handled in appstore? (link to explanations)
- BTW i appreciate the work on registration app and hope there is a way i could help...

**Comment by nickvergessen** [comment]
- Do i have to regenerate everything and PR once more? 

yes 😿 

- Planning to user twigacloud_signup for APP ID, is it ok? 

👍🏼 

- can i keep the repo private just incase?

👍🏼 

- How are private apps handled in appstore?

You don't have to put it into the appstore at all. Basically after you requested the certificate here, you stop the process.

- BTW i appreciate the work on registration app and hope there is a way i could help...

**Comment by brunoalfred** [comment]
> 

We use AIO to setup the NC instance ie we dont have access to nextcloud/apps folder, 
what is the way to install an app if it doesn't exist in app store without placing it in apps directory manually?

**Comment by szaimen** [comment]
> > 
> 
> We use AIO to setup the NC instance ie we dont have access to nextcloud/apps folder, what is the way to install an app if it doesn't exist in app store without placing it in apps directory manually?

Hi,, can you open a new discussion here: https://github.com/nextcloud/all-in-one/discussions/new?category=questions

I'll answer your question then. Thanks!

**Comment by brunoalfred** [comment]
> 

https://github.com/nextcloud/all-in-one/discussions/1287

File: signup/signup.csr [hunk 1]
```diff
@@ -0,0 +1,26 @@
+-----BEGIN CERTIFICATE REQUEST-----
```
[the remark this answers is not in the document]
Comment (brunoalfred) [reply signup/signup.csr]: 👍

**Comment by brunoalfred** [comment]
> 

The plan is to have an app that would send verification code using our internal gateway to some sort of security policies. The app wont be publicly available on the store by maybe the repository will remain open. 

The plan is to use registration app as an inspiration/reference. 

PS: im not quite familiar with [TwoFactor Gateway](https://github.com/nextcloud/twofactor_gateway/) and [playSMS](https://playsms.org/) but i'll look onto them.

**Review by vitormattos — APPROVED** [review]

**Comment by vitormattos** [comment]
Your crt here: 6cdbedcc87f21b159bf2d9024053ac591dd7d06d

**Pull request merged**